tests/report.py

```diff
@@ -50,23 +55,26 @@ def gen_report():
 
     
     report["scenarios"] = {name: {} for name in os.listdir("scenarios")}
-    for scenario in scenarios:
-        scenario_spec = {
-            "pyciemss": os.listdir(f"scenario/{scenario}/pyciemss"),
-            "sciml": os.listdir(f"scenario/{scenario}/sciml")
-        }
-        for service_name, tests in scenario_spec:
+    for scenario in report["scenarios"]:
+        scenario_spec = {}
+        for backend in ["pyciemss", "sciml"]:
+            path = f"scenarios/{scenario}/{backend}"
+            if os.path.exists(path):
+                scenario_spec[backend] = os.listdir(f"scenarios/{scenario}/{backend}")
+        for service_name, tests in scenario_spec.items():
             for test_file in tests:
-                test = test_file.strip(".json")
+                test = test_file.split(".")[0]
```

Comment:
This is fine, for reals. But generally for this use case, I prefer `os.path.splitext()`. Slightly more safe/portable and clearly conveys intent.